Doesn't build on linux

You get a lot of linker errors like 

```
/usr/bin/ld: obj/cengine.o:/home/src/src/unix/Corange/./include/cengine.h:80: multiple definition of `debug_str'; obj/casset.o:/home/src/src/unix/Corange/./include/cengine.h:80: first defined here
/usr/bin/ld: obj/cengine.o:/home/src/src/unix/Corange/./include/cengine.h:79: multiple definition of `debug_buf'; obj/casset.o:/home/src/src/unix/Corange/./include/cengine.h:79: first defined here
```

The problem is that the `cengine.h` header defines global buffer variables, that are compiled into each file:

``` c
#define ERROR_BUFFER_SIZE   2048*4
#define DEBUG_BUFFER_SIZE   2048*4
#define WARNING_BUFFER_SIZE 2048*4

...

char error_buf[ERROR_BUFFER_SIZE];
char error_str[ERROR_BUFFER_SIZE];

char warning_buf[WARNING_BUFFER_SIZE];
char warning_str[WARNING_BUFFER_SIZE];

char debug_buf[DEBUG_BUFFER_SIZE];
char debug_str[DEBUG_BUFFER_SIZE];
```

There are two quick solutions

* turn them into declarations with `extern` in `cengine.h` and define them in `cengine.c` once
* make these buffers local variables to the block, like so:

``` c
#define error(MSG, ...) { \  
  char error_buf[ERROR_BUFFER_SIZE]; \
  char error_str[ERROR_BUFFER_SIZE]; \
  snprintf(error_str,(ERROR_BUFFER_SIZE-1), "[ERROR] (%s:%s:%i) ", __FILE__, __func__, __LINE__); \
  snprintf(error_buf,(ERROR_BUFFER_SIZE-1), MSG, ##__VA_ARGS__); strcat(error_str, error_buf);    \
  error_(error_str); }
```
This is thread-safer as a bonus, but putting 16K on the stack seems somewhat excessive.

--- a/include/cengine.h
+++ b/include/cengine.h
@@ -70,14 +70,14 @@ void warning_(const char*);
 void debug_(const char*);
 
 
-char error_buf[ERROR_BUFFER_SIZE];
-char error_str[ERROR_BUFFER_SIZE];
+extern char error_buf[ERROR_BUFFER_SIZE];
+extern char error_str[ERROR_BUFFER_SIZE];
 
-char warning_buf[WARNING_BUFFER_SIZE];
-char warning_str[WARNING_BUFFER_SIZE];
+extern char warning_buf[WARNING_BUFFER_SIZE];
+extern char warning_str[WARNING_BUFFER_SIZE];
 
-char debug_buf[DEBUG_BUFFER_SIZE];
-char debug_str[DEBUG_BUFFER_SIZE];
+extern char debug_buf[DEBUG_BUFFER_SIZE];
+extern char debug_str[DEBUG_BUFFER_SIZE];
 
 #define error(MSG, ...) { \
   snprintf(error_str,(ERROR_BUFFER_SIZE-1), "[ERROR] (%s:%s:%i) ", __FILE__, __func__, __LINE__); \

--- a/src/cengine.c
+++ b/src/cengine.c
@@ -1,5 +1,14 @@
 #include "cengine.h"
 
+char error_buf[ERROR_BUFFER_SIZE];
+char error_str[ERROR_BUFFER_SIZE];
+
+char warning_buf[WARNING_BUFFER_SIZE];
+char warning_str[WARNING_BUFFER_SIZE];
+
+char debug_buf[DEBUG_BUFFER_SIZE];
+char debug_str[DEBUG_BUFFER_SIZE];
+
 fpath P(const char* path) {
   fpath p;
   

